### PR TITLE
Make tech chips clickable and sort by chips

### PR DESCRIPTION
**Pull request by Smith2106**
This pull request is for issue #35 to sort when tech chips are clicked. It works on top of the search function.

### CODE REVIEW

**Comment by TheWillG** [comment]
Thanks, will review today

**Comment by TheWillG** [comment]
@Smith2106 can you please add a filter to remove the projects on the table that do not have the tech word selected? In other words, if I click NodeJS, I only want to see projects in the table that have NodeJS listed as a tech. 
Thanks!

**Comment by TheWillG** [comment]
@Smith2106 Concerning the UI, this is exactly what I was looking for, great work and solid contribution!

**Comment by Smith2106** [comment]
@TheWillG I add a filter to remove the projects that don't have the clicked tech chips. Let me know if that is what you were looking for!

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -108,21 +112,38 @@ class Home extends Component {
     this.setState({ modalOpen: true });
   }
 
+  toggleTechChip(event) {
+    const { techChips } = this.state;
+    const tech = event.target.textContent;
+    if (!techChips.includes(tech)) techChips.push(tech);
```
Comment (TheWillG) [inline src/components/Home/Home.js]: Please use braces {} on conditional if-else statements

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -108,21 +112,38 @@ class Home extends Component {
...
-      const query = event.target.value.toLowerCase();
       const tableContent = `${item.name}${item.description}${item.tech}`;
+      let hasTechChip = true;
+      
+      this.state.techChips.forEach((techChip) => {
+        if (!item.tech.includes(techChip)) hasTechChip = false;
+      });
 
-      return tableContent.toLowerCase().indexOf(query) >= 0;
+      if (this.state.query !== '') return tableContent.toLowerCase().indexOf(this.state.query) >= 0 && hasTechChip;
```
Comment (TheWillG) [inline src/components/Home/Home.js]: Braces { }

File: src/components/Home/Home.js [hunk 3]
```diff
@@ -108,21 +112,38 @@ class Home extends Component {
...
+  filterProjects() {
     const { initialProjects } = this.state;
     let updatedProject = initialProjects;
     updatedProject = updatedProject.filter((item) => {
-      const query = event.target.value.toLowerCase();
       const tableContent = `${item.name}${item.description}${item.tech}`;
+      let hasTechChip = true;
+      
+      this.state.techChips.forEach((techChip) => {
+        if (!item.tech.includes(techChip)) hasTechChip = false;
```
Comment (TheWillG) [inline src/components/Home/Home.js]: Braces { }

File: src/components/Home/Home.js [hunk 4]
```diff
@@ -108,21 +112,38 @@ class Home extends Component {
...
     this.setState({ modalOpen: false });
-    this.fetchProjects();
+    this.filterProjects();
   }
 
-  filterProjects(event) {
+  filterProjects() {
     const { initialProjects } = this.state;
     let updatedProject = initialProjects;
     updatedProject = updatedProject.filter((item) => {
```
Comment (TheWillG) [inline src/components/Home/Home.js]: Please use `project` instead of `item`. It is more descriptive

**Comment by TheWillG** [comment]
@Smith2106  One more thing I noticed, the sorting by newest to oldest is no longer working when clicking the chips and then unclicking them. We want to make sure order of projects is always newest on top and oldest on bottom. Thanks again for your help.

**Comment by Smith2106** [comment]
I fixed all the small issues and now it always sorts from newest to oldest.

**Comment by TheWillG** [comment]
@Smith2106 looks great, thanks for the contribution!

**Pull request merged**